### PR TITLE
Merge esc

### PR DESCRIPTION
**Pull request by lenter83**

### CODE REVIEW

File: src/escd/network/peerclientmanager.cpp [hunk 1]
```diff
@@ -266,92 +276,82 @@ void PeerConnectManager::connect(std::string peer_address, unsigned short port,
...
+            boost::asio::ip::tcp::resolver::query       query(peer_address.c_str(),SERVER_PORT);
+            boost::asio::ip::tcp::resolver::iterator    iterator = resolver.resolve(query);
+            boost::asio::ip::tcp::resolver::iterator    end;
 
-        connect(peer_address, atoi(port.c_str()), atoi(svid.c_str()));
-        --connNeeded;
+
+            if(iterator != end)
+            {
+               if(inet_aton(iterator->endpoint().address().to_string().c_str(), &addr) != 0)
```
Comment (adshares-tech) [inline src/escd/network/peerclientmanager.cpp]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)

File: src/escd/network/peerclientmanager.cpp [hunk 2]
```diff
@@ -266,92 +276,82 @@ void PeerConnectManager::connect(std::string peer_address, unsigned short port,
...
+void PeerConnectManager::getPeersFromDNS(std::vector<std::pair<in_addr_t, unsigned short>> &peerAddrs)
 {
     try
     {
         boost::asio::ip::tcp::resolver              resolver(m_ioService);
         boost::asio::ip::tcp::resolver::query       query(m_opts.dnsa.c_str(),SERVER_PORT);
         boost::asio::ip::tcp::resolver::iterator    iterator = resolver.resolve(query);
         boost::asio::ip::tcp::resolver::iterator    end;
-        std::map<uint32_t,boost::asio::ip::tcp::resolver::iterator> endpoints;
+        std::vector<boost::asio::ip::tcp::resolver::iterator> endpoints;
```
Comment (adshares-tech) [inline src/escd/network/peerclientmanager.cpp]: MINOR Remove this unused "endpoints" local variable. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1481)

**Comment by adshares-tech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [peerclientmanager.h#L136](https://github.com/adshares/ads/blob/8adef608efe5ae9d0df6b26fc5fdb8be6c7d3e22/src/escd/network/peerclientmanager.h#L136): Remove this unused "m_sourceCounter" "private" field. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1068)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/escd/network/peerclientmanager.cpp [hunk 3]
```diff
@@ -266,92 +277,81 @@ void PeerConnectManager::connect(std::string peer_address, unsigned short port,
...
-            //Don't connect to myself
-            if(nodeIndx != m_opts.svid && m_server.getNode(nodeIndx, nodeInfo))
-            {
-                if(nodeInfo.port > 0
-                        && m_peers.find(std::make_pair(nodeInfo.ipv4, nodeInfo.port)) == m_peers.end() )
-                {
-                    foundNew = true;
-                    break;
-                }
+            if(nodeInfo.mtim >= m_server.last_srvs_.now - BLOCKSEC*BLOCKDIV) {
```
Comment (adshares-tech) [inline src/escd/network/peerclientmanager.cpp:351]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)

**Comment by adshares-tech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by m-pilarczyk — APPROVED** [review]

**Pull request merged**